New Playbook - Update-CVE-IPs-WatchListwithGreyNoise

swiftsolves-msft opened the pull request:
This playbook uses the GreyNoise API to search for intresting IPs discovered in the last day tagged per each CVE found in the mode you setup. See below for more information on the Modes to deploy with. This is a daily timer  trigger to go out and fetch things and push to a watchlist, also comes with a sample detection analytic rule.

   Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below


# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)


-----------------------------------------------------------------------------------------------------------

v-spadarthi commented:
@manishkumar1991 / @rahul0216 : Please have a look and provide your feedback. Thanks!

v-spadarthi commented:
@manishkumar1991 / @rahul0216 : Please have a look and provide your feedback. Thanks!

rahul0216 commented on Playbooks/Update-CVE-IPs-WatchListwithGreyNoise/azuredeploy.json:
```diff
@@ -0,0 +1,1301 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
```
metadata requires additional fields. Look at this playbook's metadata section for reference https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/OpenCTI/Playbooks/OpenCTIPlaybooks/OpenCTI-CreateIndicator/azuredeploy.json

rahul0216 commented on Playbooks/Update-CVE-IPs-WatchListwithGreyNoise/azuredeploy.json:
```diff
@@ -0,0 +1,1301 @@
+        "GreyNoiseAPIKey": {
+            "type": "securestring"
+        },
+        "CVEMode": {
```
Use allowedValues to restrict the CVEMode options. 
https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/parameters#allowed-values

rahul0216 reviewed (commented):
1. Please address the inline comments.
2. Add images of Playbook in light and dark theme. 
3. There is a Test branch in the Playbook. Not sure it that is intended or left by mistake.

v-spadarthi commented:
@swiftsolves-msft : Please address the @rahul0216 comments.

v-mchatla commented:
Hi @swiftsolves-msft,
 Please address the @rahul0216 comments.
Thanks

swiftsolves-msft commented:
> 1. Please address the inline comments.
> 2. Add images of Playbook in light and dark theme.
> 3. There is a Test branch in the Playbook. Not sure it that is intended or left by mistake.

@rahul0216 \ @v-mchatla 

Thanks - addressed comments in latest syntax updates pull request by making the asked modifications in readme.md and azuredeploy.json. Added image of playbook logic is in readme file. The test branch can remain in Logic App as it can be useful for folks if the need arises for debugging but does not execute unless switched too explicitly.

Thanks,
Nathan

swiftsolves-msft commented:
> SWITCH is happening on CVEList Parameter, but it should be based on CVEMode parameter. Can you please test the playbook once again? ![image](https://user-images.githubusercontent.com/8243910/208434468-db9048e9-d719-4307-916a-309471b1e457.png)

The following attached is the reasoning, however if you think the design needs to be simplified I can remove the Global Param for 'CVEMode' altogehter and in ARM Template Deployment update the action 2 the variable instead with the choosen values. Please feel free to contact me also on teams internally if needed ~ Nathan Swift

![image](https://user-images.githubusercontent.com/30509195/208526156-25a5c987-7f9c-4061-8938-bbe0e78f6513.png)

rahul0216 reviewed (approved):
Thanks Nathan for clarification. Changes looks good.

v-spadarthi reviewed (approved):
Rahul tested the changes and approved for another approval required giving approval
